packages/@aws-cdk/cdk/lib/core/tokens.ts

```diff
@@ -169,27 +197,31 @@ export function resolve(obj: any, prefix?: string[]): any {
     return obj;
   }
 
-  //
-  // tokens - invoke 'resolve' and continue to resolve recursively
-  //
-
-  if (unresolved(obj)) {
-    const value = obj[RESOLVE_METHOD]();
-    return resolve(value, path);
-  }
-
   //
   // arrays - resolve all values, remove undefined and remove empty arrays
   //
 
   if (Array.isArray(obj)) {
+    if (containsListToken(obj)) {
```

Comment:
No, the issue is to detect the case where someone has done:

```ts
const xs = token.toList();
xs.push('hello');

// Or
const xs = ['hello'].concat(token.toList());
```

In that case, we want to detect that the token as constructed was meant to be a Token but is now illegal because of the concatenation.